src/extensions/extensionInterface.ts

```diff
@@ -211,7 +211,39 @@ type SetVisibleUnitIdsSortingSelectionAction = {
     visibleUnitIds: number[] | null
 }
 
-export type SortingSelectionAction = SetSelectionSortingSelectionAction | SetSelectedUnitIdsSortingSelectionAction | SetVisibleUnitIdsSortingSelectionAction
+type UnitClickedSortingSelectionAction = {
+    type: 'UnitClicked'
+    unitId: number
+    ctrlKey?: boolean
+    shiftKey?: boolean
+}
+
+export type SortingSelectionAction = SetSelectionSortingSelectionAction | SetSelectedUnitIdsSortingSelectionAction | SetVisibleUnitIdsSortingSelectionAction | UnitClickedSortingSelectionAction
+
+const unitClickedReducer = (state: SortingSelection, action: UnitClickedSortingSelectionAction): SortingSelection => {
+    const unitId = action.unitId
+    if (action.ctrlKey) {
+        if ((state.selectedUnitIds || []).includes(unitId)) {
+            return {
+                ...state,
+                selectedUnitIds: (state.selectedUnitIds || []).filter(uid => (uid !== unitId))
+            }
+        }
+        else {
+            return {
+                ...state,
+                selectedUnitIds: [...(state.selectedUnitIds || []), unitId]
+            }
+        }
+    }
+    // todo: restore anchor/shift-select behavior somewhere
```

Comment:
Will need to restore shift-click range select functionality somewhere. (maybe not in the reducer because we need to know the list of unit ids)

Reply:
Hmmm... yeah, since they can be discontinuous, without knowing the exact set of unit IDs it can be annoying to try to handle range select. I do recall at some point or another I handled this by just throwing the entire newly selected range into the set of selected Unit IDs and letting the front-end figure out if the corresponding unit IDs even existed (...and of course I spent several hours tracking down bugs that resulted from referencing selected unit IDs that didn't exist when I forgot to handle it properly, so maybe that isn't a productive reminiscence).

Would it be at all practical to design the `unitClickedReducer` to store the state on the set of unit IDs (implying, for it to actually be initialized at some point with the full list of IDs that exist)?

As a separate thought on the topic of range selection, we had been handling discontinuous ranges of IDs but we were always assuming that the unit IDs would be displayed on screen in ascending order. If, for some reason, that was not the case, then the old range select would break anyway. (I don't know if this is at all likely or possible, but I consider thinking of implausible scenarios part of my job.) So if & when we reimplement range select, it might be beneficial to have a list that actually preserves the ordering of the unit IDs.

Actually I suppose unit IDs could very easily be reordered in the units table since we have sorting functionality; while selection in that context is handled by check boxes, I wonder if we shouldn't coordinate display in other views with whatever ordering is currently active in the units table... so that e.g. an autocorrelograms view would resort the charts if the user re-sorts the table?